include/aikido/util/detail/yaml_extension.hpp

```diff
@@ -135,13 +131,13 @@ struct convert<Eigen::
       throw YAML::RepresentationException(getMark(node), ss.str());
     }
 
-    if (node.Tag() == "!Vector")
+    if (node[0].Type() == YAML::NodeType::Scalar)
     {
       matrix.resize(rows, 1);
       for (Index i = 0; i < rows; ++i)
         matrix(i, 0) = node[i].template as<_Scalar>();
```

Comment:
> Yes, we have to handle the case where node doesn't have a child. I believe the current implementation will SEGFAULT because we access node[0].Type() even when node.size() == 0.

I confirmed that this case is handled by [this](https://github.com/personalrobotics/aikido/pull/200/files#diff-ee420700678d3f6c35f1b8428db1bd8eR118) through a [test](https://github.com/personalrobotics/aikido/pull/200/files#diff-87b50c6224fda02940d3d364f0606c50R10). If the node doesn't have a child, then the node type is `Null`.

> I also agree agree with @psigen - we should have an additional check that node[i].Type() == YAML::NodeType::Scalar before we try the conversion node[i].template as<_Scalar>().
>
> @jslee02 I saw your link, but don't understand why the tests are relevant here. :thinking:

I thought he meant incorrect column size for each row. The suggested additional checks are added for both of vector and matrix cases.